src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/eventProcessor/EventProcessor.scala

```diff
@@ -43,7 +43,8 @@ class EventProcessor(eventRepository: EventRepository,
         dagDefinitionRepository.getJoinedDagDefinition(properties.sensorId).flatMap {
           case Some(joinedDagDefinition) =>
             for {
-              dagInstanceJoined <- jobTemplateService.resolveJobTemplate(joinedDagDefinition, triggeredBy)
+              hasRunningDagInstance <- dagInstanceRepository.hasRunningDagInstance(joinedDagDefinition.workflowId)
```

Comment:
I would change this to hasInQueue. That one will be running, one in queue and rest will be skipped

Reply:
Hmm. `DagInstanceRepository.hasRunningDagInstance` selects statuses `Running` and `InQueue`. So `hasRunningDagInstance` and `hasInQueue` are equally imprecise. I could rename to `hasNonFinal`, but I'm not sure if that would be better

Reply:
As discussed: 
When you have 5 jobs created one after another you will end up with:
Running, InQueue, Skipped, Skipped, Skipped
Because there can be really long execution of instance but you receive another event so that one will go to queue and if you receive another, that one will be merged/skipped with that which is already in queue